dlog/circuits/src/compiled.rs

```diff
@@ -114,11 +121,14 @@ impl<G: AffineCurve> Compiled<G>
         oracle2: Fr<G>,
     ) -> Vec<Fr<G>>
     {
-        self.row_eval_b.evals.iter().zip(self.col_eval_b.evals.iter()).map
+        self.row_eval_b.evals.iter().
+            zip(self.col_eval_b.evals.iter()).
+            zip(self.rc_eval_b.evals.iter()).
+            map
         (
-            |(row, col)|
+            |((row, col), rc)|
```

Comment:
I think the previous implementation of this way more efficient (it used only 1 multiplication rather than the 3 used here).

Reply:
But they're not equivalent as polynomials though. Letting `f,g` be defined as follows:
`f(x) = (row(x) - alpha) * (col(x) - beta)`
`g(x) = row(x)*col(x) - alpha * col(x) - beta*row(x) + alpha*beta`

`f(x) != g(x)` for any x not in `K`. `x` can't be in `K` due to zero-knowledge (and completeness IIRC?)

Reply:
Sorry, you're right -- I got confused and thought it was just evaluating over a domain where they agreed.